--- origsrc/xf86-video-qxl/src/qxl.h	2014-01-09 19:28:49.000000000 -0600
+++ src/xf86-video-qxl/src/qxl.h	2014-01-09 22:39:19.427710300 -0600
@@ -146,7 +146,9 @@ enum {
     OPTION_SPICE_PLAYBACK_FIFO_DIR,
     OPTION_SPICE_VDAGENT_ENABLED,
     OPTION_SPICE_VDAGENT_VIRTIO_PATH,
+#ifdef __linux__
     OPTION_SPICE_VDAGENT_UINPUT_PATH,
+#endif
     OPTION_SPICE_VDAGENT_UID,
     OPTION_SPICE_VDAGENT_GID,
 #endif
--- origsrc/xf86-video-qxl/src/qxl_driver.c	2014-01-09 19:28:49.000000000 -0600
+++ src/xf86-video-qxl/src/qxl_driver.c	2014-01-09 22:39:57.591893200 -0600
@@ -69,8 +69,10 @@ static char filter_str[] = "filter";
 static char auto_str[]   = "auto";
 static char auto_glz_str[]   = "auto_glz";
 static char spice_vdagent_virtio_path_default[] = "/tmp/xspice-virtio";
+#ifdef __linux__
 static char spice_vdagent_uinput_path_default[] = "/tmp/xspice-uinput";
 #endif
+#endif
 static char driver_name[] = QXL_DRIVER_NAME;
 const OptionInfoRec DefaultOptions[] =
 {
@@ -140,8 +142,10 @@ const OptionInfoRec DefaultOptions[] =
       "SpiceVdagentEnabled",      OPTV_BOOLEAN,   {0}, FALSE},
     { OPTION_SPICE_VDAGENT_VIRTIO_PATH,
       "SpiceVdagentVirtioPath",   OPTV_STRING,    {.str = spice_vdagent_virtio_path_default}, FALSE},
+#ifdef __linux__
     { OPTION_SPICE_VDAGENT_UINPUT_PATH,
       "SpiceVdagentUinputPath",   OPTV_STRING,    {.str = spice_vdagent_uinput_path_default}, FALSE},
+#endif
     { OPTION_SPICE_VDAGENT_UID,
       "SpiceVdagentUid",          OPTV_INTEGER,    {0}, FALSE},
     { OPTION_SPICE_VDAGENT_GID,
--- origsrc/xf86-video-qxl/src/spiceqxl_uinput.c	2014-01-09 19:28:49.000000000 -0600
+++ src/xf86-video-qxl/src/spiceqxl_uinput.c	2014-01-09 22:37:44.797297700 -0600
@@ -1,3 +1,5 @@
+#ifdef __linux__
+
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
@@ -124,3 +126,5 @@ void spiceqxl_uinput_init(qxl_screen_t *
     qxl->core->watch_add(uinput_fd, SPICE_WATCH_EVENT_READ, spiceqxl_uinput_read_cb, qxl);
     spice_server_set_agent_mouse(qxl->spice_server, 1);
 }
+
+#endif /* __linux__ */
--- origsrc/xf86-video-qxl/src/spiceqxl_vdagent.c	2014-01-09 19:28:49.000000000 -0600
+++ src/xf86-video-qxl/src/spiceqxl_vdagent.c	2014-01-09 22:38:40.501483800 -0600
@@ -11,7 +11,9 @@
 #include "qxl_option_helpers.h"
 
 #include "spiceqxl_util.h"
+#ifdef __linux__
 #include "spiceqxl_uinput.h"
+#endif
 #include "spiceqxl_vdagent.h"
 
 static const char *vdagent_virtio_filename;
@@ -168,5 +170,7 @@ void spiceqxl_vdagent_init(qxl_screen_t
 
     vdagent_sin.base.base.sif = &vmc_interface.base;
     spice_server_add_interface(qxl->spice_server, &vdagent_sin.base.base);
+#ifdef __linux__
     spiceqxl_uinput_init(qxl);
+#endif
 }
