--- origsrc/xf86-video-qxl-0.0.17/src/Makefile.am	2012-03-15 12:47:41.000000000 -0500
+++ src/xf86-video-qxl-0.0.17/src/Makefile.am	2012-07-01 02:41:15.192772000 -0500
@@ -34,7 +34,7 @@ qxl_drv_la_LTLIBRARIES = qxl_drv.la
 qxl_drv_la_LDFLAGS = -module -avoid-version
 qxl_drv_ladir = @moduledir@/drivers
 
-qxl_drv_la_LIBADD = uxa/libuxa.la
+qxl_drv_la_LIBADD = uxa/libuxa.la $(XORG_LIBS)
 
 qxl_drv_la_SOURCES =				\
 	qxl.h					\
@@ -61,7 +61,7 @@ spiceqxl_drv_ladir = @moduledir@/drivers
 
 spiceqxl_drv_la_CFLAGS = -DXSPICE $(AM_CFLAGS) $(SPICE_CFLAGS)
 
-spiceqxl_drv_la_LIBADD = uxa/libuxa.la
+spiceqxl_drv_la_LIBADD = uxa/libuxa.la $(XORG_LIBS)
 
 spiceqxl_drv_la_SOURCES =				\
 	qxl.h					\
